File: pascal/types/strings/string_interpolations.pas
program StringInterpolations;

uses
  SysUtils;

var
  i: integer = 1;
  f: real = 2.3;
  s: string = 'apple';
  a: array[0..2] of string = (
    'orange',
    'blue',
    'green'
  );

begin
  // https://docwiki.embarcadero.com/Libraries/Alexandria/en/System.SysUtils.Format
  writeln(Format('i = %d, f = %f, s = %s', [i, f, s]));
  writeln(Format('a[0] = %s, a[1] = %s, a[2] = %s', [a[0], a[1], a[2]]));
end.

File: c++/types/strings/string_interpolations.cpp
#include <iostream>
#include <string>

int main()
{
  int i = 1;
  float f = 2.3;
  std::string s = "apple";

  printf("i = %d, f = %f, s = %s\n\n", i, f, s.c_str());

  std::string a[] = {
    "orange",
    "blue",
    "green"
  };

  printf("a[0] = %s, a[1] = %s, a[2] = %s\n\n", a[0].c_str(), a[1].c_str(), a[2].c_str());
}
